weather.py

```diff
@@ -0,0 +1,182 @@
+        "Mean Humidity", weather_data, lambda x, y: x > y, highest_humidity, highest_humidity_day)
+
+    return {
+        "Highest": (highest_temperature, highest_temprature_day),
+        "Lowest": (lowest_temperature, lowest_temprature_day),
+        "Humidity": (highest_humidity, highest_humidity_day)
+    }
+
+
+def calculate_monthly_averages(weather_data):
+    def calculate_average(key):
+        values = [
+            int(daily_data[key])
+            for daily_data in weather_data if daily_data.get(key)
+        ]
+        return sum(values) / len(values)
+
+    avg_high_temprature = calculate_average("Max TemperatureC")
+    avg_low_temprature = calculate_average("Min TemperatureC")
+    avg_mean_humidity = calculate_average("Mean Humidity")
+
+    return {
+        "Average Highest Temperature": avg_high_temprature,
+        "Average Lowest Temperature": avg_low_temprature,
+        "Average Mean Humidity": avg_mean_humidity
+    }
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("directory", type=str)
+    parser.add_argument("-e", "--year", type=int)
+    parser.add_argument("-a", "--year_month", type=str)
+    parser.add_argument("-c", "--chart", type=str,)
+    return parser.parse_args()
+
+
+def print_yearly_stats(stats):
+    high_temprature, high_day = stats["Highest"]
+    low_temprature, low_day = stats["Lowest"]
+    humid_val, humid_day = stats["Humidity"]
+
+    date_high = high_day.split("-")
+    date_low = low_day.split("-")
+    date_humid = humid_day.split("-")
+
+    print(f"Highest: {high_temprature}C on {calendar.month_abbr[int(date_high[1])]} {date_high[2]}")
+    print(f"Lowest: {low_temprature}C on {calendar.month_abbr[int(date_low[1])]} {date_low[2]}")
+    print(f"Humidity: {humid_val}% on {calendar.month_abbr[int(date_humid[1])]} {date_humid[2]}")
+
+
+def print_monthly_stats(stats):
+    print(f"Highest Average Temperature: {stats["Average Highest Temperature"]:.0f}C")
+    print(f"Lowest Average Temperature: {stats["Average Lowest Temperature"]:.0f}C")
+    print(f"Average Mean Humidity: {stats["Average Mean Humidity"]:.0f}%")
+
+
+def create_temperature_bars(day_data):
+    max_temprature = 0
+    min_temprature = 0
+    if day_data["Max TemperatureC"]:
+        max_temprature = int(day_data["Max TemperatureC"])
+        min_temprature = int(day_data["Min TemperatureC"])
+    max_bar = "+" * max_temprature
+    min_bar = "+" * min_temprature
+
+    return max_bar, min_bar, max_temprature, min_temprature
+
+
+def display_weather_chart(month, year, weather_data):
+    RED = "\033[31m"
+    BLUE = "\033[34m"
+    RESET = "\033[0m"
+
+    print(calendar.month_abbr[month], year)
+    for day_data in weather_data:
+        max_temp_bar , min_temp_bar, max_temprature, min_temprature = create_temperature_bars(day_data)
+        day = day_data.get("PKT", day_data.get("PKST")).split("-")[2]
+
+        if max_temprature != 0:

```

Comment:
For some cities in world max temperature can be 0, what is the rationale of such check?

Reply:
Kindly look into spellings carefully.